### PR TITLE
Nagios host env

### PR DESCRIPTION
**Pull request by SwannCroiset**
Suffix Nagios host with the environment_label dimension

### CODE REVIEW

File: heka/files/lua/encoders/status_nagios.lua [hunk 1]
```diff
@@ -64,10 +66,17 @@ function process_message()
...
 
+    local host
     if host_dimension_key then
-        data['host'] = read_message(string.format('Fields[%s]', host_dimension_key)) or default_host
+        host = read_message(string.format('Fields[%s]', host_dimension_key)) or default_host
+    else
+        host = read_message('Fields[hostname]') or read_message('Hostname')
+    end
+
+    if host_suffix_dimension_key then
```
Comment (simonpasquier) [inline heka/files/lua/encoders/status_nagios.lua]: (nit) I would read `Fields[<host_suffix_dimension_key>]` first and only append if it's not nil.

Comment (simonpasquier) [reply heka/files/lua/encoders/status_nagios.lua]:
```
if host_suffix_dimension_field then
  local suffix = read_message(host_suffix_dimension_field)
  if suffix then
    host = host .. '.' .. suffix
  end
end
data['host'] = host
```

File: heka/files/lua/encoders/status_nagios.lua [hunk 2]
```diff
@@ -19,6 +19,8 @@ local consts = require 'gse_constants'
 local lma = require 'lma_utils'
 local interp = require "msg_interpolate"
 
+local host_suffix_dimension_key = read_config('host_suffix_dimension_key')
```
Comment (simonpasquier) [inline heka/files/lua/encoders/status_nagios.lua]: nit but you could build `Fields[...]` just once here:

```
local host_suffix_dimension_field
if read_config('host_suffix_dimension_key') then
  host_suffix_dimension_field = string.format('Fields[%s]', read_config('host_suffix_dimension_key'))
end
```

**Review by simonpasquier — CHANGES_REQUESTED** [review]

File: heka/files/lua/encoders/status_nagios.lua (hunk 2 above)
Comment (SwannCroiset) [reply heka/files/lua/encoders/status_nagios.lua]: done

File: heka/files/lua/encoders/status_nagios.lua (hunk 1 above)
Comment (SwannCroiset) [reply heka/files/lua/encoders/status_nagios.lua]: done

File: heka/files/lua/encoders/status_nagios.lua [hunk 3]
```diff
@@ -19,9 +19,17 @@ local consts = require 'gse_constants'
 local lma = require 'lma_utils'
 local interp = require "msg_interpolate"
 
+local host_suffix_dimension_field
+if read_config('host_suffix_dimension_key') then
+    host_suffix_dimension_field = string.format('Fields[%s]', read_config('host_suffix_dimension_key'))
+end
```
Comment (elemoine) [inline heka/files/lua/encoders/status_nagios.lua:25]: (nit) You could avoid reading the config twice. Up to you.

```lua
local host_suffix_dimension_field
local host_suffix_dimension_key = read_config('host_suffix_dimension_key')
if host_suffix_dimension_key then
     host_suffix_dimension_field = string.format('Fields[%s]', host_suffix_dimension_key)
end
```

File: heka/files/lua/encoders/status_nagios.lua [hunk 4]
```diff
@@ -19,9 +19,17 @@ local consts = require 'gse_constants'
...
+
 -- These 2 configurations are used only to encode GSE messages
 local default_host = read_config('default_nagios_host')
-local host_dimension_key = read_config('nagios_host_dimension_key')
+local host_dimension_field
+if read_config('nagios_host_dimension_key') then
+    host_dimension_field = string.format('Fields[%s]', read_config('nagios_host_dimension_key'))
+end
```
Comment (elemoine) [inline heka/files/lua/encoders/status_nagios.lua:32]: Ditto.

**Review by elemoine — APPROVED** [review]
With "nit" comments.

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**